src/map/utils/zoneutils.cpp

```diff
@@ -984,6 +984,9 @@ int GetWeatherElement(WEATHER weather)
 {
     TPZ_DEBUG_BREAK_IF(weather >= MAX_WEATHER_ID);
 
+    // TODO: Fix weather ordering; at the moment, this current fire, water, earth, wind, snow, thunder
+    // order MUST be preserved due to the weather enums going in this order. Those enums will
+    // most likely have rippling effects, such as how weather data is stored in the db
```

Comment:
I didn't change the ordering in this array due to the fact that weather is currently in an... odd... order. You might have to expand down to see it. This is in both core [and lua](https://github.com/ibm2431/topaz/blob/master/scripts/globals/weather.lua#L6-L23).

This may be of interest to @Kreidos

Reply:
The WEATHER enum stores the actual packet/client values for weather so the order there is accurate and should probably stay unchanged. As well, nothing that changes here should affect weather in-game or in the DB as those only rely on the WEATHER enum being correct. :)

Reply:
The WEATHER enum stores the actual packet/client values for weather so the order there is accurate and should probably stay unchanged. As well, nothing that changes here should affect weather in-game or in the DB as those only rely on the WEATHER enum being correct. :)

Reply:
Huh. I guess [I should have checked](https://github.com/Windower/Resources/blob/master/resources_data/weather.lua#L3-L23). 

Come on SE, why do you do such things to us. 😢 

I'll amend the comment - and maybe place a few others in weather enums - so others don't make the same assumption I did.

Thanks, Kreidos~